crates/sui-graphql-rpc/src/types/transaction_block_effects.rs

```diff
@@ -93,11 +103,46 @@ impl TransactionBlockEffects {
     }
 
     /// Transactions whose outputs this transaction depends upon.
-    async fn dependencies(&self, ctx: &Context<'_>) -> Result<Option<Vec<TransactionBlock>>> {
-        ctx.data_unchecked::<PgManager>()
-            .fetch_txs_by_digests(self.native.dependencies())
+    async fn dependencies(
+        &self,
+        ctx: &Context<'_>,
+        first: Option<u64>,
+        after: Option<CDependencies>,
+        last: Option<u64>,
+        before: Option<CDependencies>,
+    ) -> Result<Connection<String, TransactionBlock>> {
+        let page = Page::from_params(ctx.data_unchecked(), first, after, last, before)?;
+        let mut connection = Connection::new(false, false);
+
+        let dependencies = self.native.dependencies();
+
+        let Some((prev, next, cs)) = page.paginate_indices(dependencies.len()) else {
+            return Ok(connection);
+        };
+
+        let indices: Vec<CDependencies> = cs.map(|c| c).collect();
```

Comment:
You should be able to omit the `map` here.

```suggestion
        let indices: Vec<CDependencies> = cs.collect();
```